File: javascript/grammer/03_string_slicing.js
const word = 'algorithm';

console.log('원본:', word);
console.log('word.slice(1, 5) ->', word.slice(1, 5));
console.log('word.slice(0, 4) ->', word.slice(0, 4));
console.log('2칸씩 건너뛰기 ->', [...word].filter((_, i) => i % 2 === 0).join(''));
console.log('뒤집기 ->', [...word].reverse().join(''));

File: java/grammer/03_string_slicing.java
class GN03StringSlicing {
    public static void main(String[] args) {
        String text = "algorithm";
        System.out.println("앞 4글자: " + text.substring(0, 4));
        System.out.println("뒤 3글자: " + text.substring(text.length() - 3));
        System.out.println("역순: " + new StringBuilder(text).reverse());
    }
}
